### DISCUSSION TITLE
Cannot pass files together with input directory

### DISCUSSION THREAD

**Discussion by noraj** [marp-team/marp#382] [Ideas] [answered]
## Context

Command (incorrect):

```
$ marp slides.md --theme-set themes --output index.html --input-dir docs
```

Error message:

```
[ ERROR ] Cannot pass files together with input directory.
[  WARN ] Not found processable Markdown file(s).
          
Usage:
marp [options] <files...>
marp [options] -I <dir>
```

Command (correct):

```
$ marp --theme-set themes --output index.html --input-dir docs
```

Tree:

```
.
├── docs
│   ├── assets
│   ├── slides.md
│   └── themes
└── README.md
```

## Issue

I should be able to provide both an input directory `docs/` and a file `slides.md`. Hopefully without the file specific marp-cli still manage the autodetect the source file in `docs/` but what if the file had been named `SLIDES` (without extension) or `slides.mkd` (less common markdown extension)?

## Idea

Outside requiring an argparse logic change there is no reason to not support using both a file and input dir at the same time.

**Comment by yhatt** [comment, marked as answer]
Marp CLI can handle multiple inputs by using directory and glob pattern. Try this:

```
marp slides.md docs/*.md --theme-set themes
```

or simply this:

```
marp slides.md docs --theme-set themes
```

By passing directories as input, Marp CLI will find Markdown files ([`*.md`, `*.mdown`, `*.markdown`, `*.markdn`](https://github.com/marp-team/marp-cli/blob/03061ff313afb2237640e5c10d38283e6d00e5ad/src/file.ts#L13)) from specified directories.
